memory/memory_interface.ml

```diff
@@ -75,13 +78,20 @@ type errors =
 
 exception MemoryError of errors
 
-let err = Error.{
-    def = errors;
-    raiser = (fun e -> raise (MemoryError e));
-    matcher = (function
-        | MemoryError e -> Some e
-        | e -> Some (Internal_error (Printexc.to_string e)))
-  }
+let err = Error.
+    { def = errors
+    ; raiser = (fun e ->
+      let exn = MemoryError e in
+      error "%s (%s)" (Printexc.to_string exn) __LOC__;
+      raise exn)
+    ; matcher = (function
+      | MemoryError e as exn ->
+          error "%s (%s)" (Printexc.to_string exn) __LOC__;
```

Comment:
I think you need to register a pretty printer for this to work:
```
#require "xcp.memory";;
let e = Memory_interface.MemoryError (Memory_interface.Cannot_free_this_much_memory (42L, 42L));;
val e : exn =
  Memory_interface.MemoryError
   (Memory_interface.Cannot_free_this_much_memory (42L, 42L))
Printexc.to_string e;;
- : string = "Memory_interface.MemoryError(_)"

let string_of_error e = Rpcmarshal.marshal Memory_interface.errors.Rpc.Types.ty e |> Rpc.to_string;;
val string_of_error : Memory_interface.errors -> string = <fun>

 let register_memory_error () =
   Printexc.register_printer (function
    | Memory_interface.MemoryError e -> Some (Printf.sprintf "Memory_interface.Memory_error(%s)" (string_of_error e)) | _ -> None);;
val register_memory_error : unit -> unit = <fun>
register_memory_error ();;
- : unit = ()

Printexc.to_string e;;
- : string =
"Memory_interface.Memory_error([S(Cannot_free_this_much_memory);[I(42);I(42)]])"

```

Reply:
(very bare bones pretty printer, maybe using `Jsonrpc.to_string` would make it look nicer?)

Reply:
Interesting. I always assumed that `Printexc.to_string e` would be able to inspect any exception.

Reply:
Yeah I don't see why the compiler couldn't autogenerate these pretty printers, it seems it only has trouble when the argument of the exception is an exception with a variant (otherwise it is able to at least print the constructor as a number)

Reply:
Maybe we could autogenerate these in `ppx_deriving_rpc` though?

Reply:
That's fairly annoying. We can investigate what to do, it may be worth opening an issue on `ocaml-rpc` (and maybe also in Jira)

Reply:
Can you please do the same also for the other ported modules, I think v6, networkd and maybe rrd.

Reply:
I am going to port this change to the Error functor in ocaml-rpc